### DISCUSSION TITLE
HomeKit Szenen schicken nicht Nachrichten an alle Gruppenadressen

### DISCUSSION THREAD

**Discussion by jhuber** [brutella/hkknx-public#222] [Q&A]
Hallo zusammen,

inzwischen läuft meine hkknx Installation nahezu perfekt. Ich habe nur ein Problem mit den Homekit Szenen. Wenn ich eine Szene für z.B. 8 Geräte/Gruppenadressen triggere, werden vielleicht 2-3 Nachrichten geschickt. Das Verhalten ist leider indeterministisch. In der Home App werden die Kommandos richtig angezeigt, also z.B. schließen alle ausgewählten Rollläden, aber es bewegen sich nur 2-3. Alle Geräte funktionieren individuell perfekt. Lediglich die Homekit (nicht KNX!) Szenen zeigen das Verhalten.

Hat jemand eine Idee? Ohne HK Szenen wird es schwer... :-) 

Danke!

**Comment by jhuber** [comment]
Nachtrag: habe mir die Mühe gemacht es etwas genauer anzuschauen. Es geht immer dann schief, wenn ich mehr als 3 Gruppenadressen habe. D.h. Szenen mit 3 Geräten sind kein Problem. Szenen mit >3 Geräten triggern dann nur noch 2-3 Geräte.

Hilfe! :-)

**Reply by brutella** [reply]
Vielleicht liegt es daran, dass die Telegramme zu schnell hintereinander versendet werden und daher teilweise verloren gehen.

**Reply by jhuber** [reply]
Was kann ich dagegen machen? :-)

**Reply by brutella** [reply]
Ich müsste ein Limit einbauen, welches die Zeit zwischen zwei aufeinanderfolgenden Telegrammen begrenzt. Siehe → #212

**Reply by jhuber** [reply]
Okay, das macht Sinn. Beide Betas haben bei mir eher zu Probleme geführt (Statusinitialisierung fehlerhaft, braucht Kontrollfahrten, etc.). Gibt es eine Möglichkeit Szene über Scripting zu emulieren oder wären KNX Szenen der nachhaltigere Weg? 

Ist eine Version mit manuellem Rate Limitting geplant bzw. wann?

**Reply by jhuber** [reply]
Aktueller Workaround, da mir nicht klar ist, ob ich per Skript in hkknx Aktoren definieren kann: die Kurzbefehl-App von iOS lässt Limits zwischen den Telegrammen einstellen. Skaliert natürlich nicht, da sehr aufwendig, aber für ein EFH durchaus machbar.

**Comment by jhuber** [comment]
PPS: hkknx 2.4.1 auf einem RPi, Geräte sind auf iOS 16.1 bzw. Ventura.
